java/libphonenumber/src/com/google/i18n/phonenumbers/PhoneNumberUtil.java

```diff
@@ -3018,6 +3018,12 @@ private boolean checkRegionForParsing(String numberToParse, String defaultRegion
    * is actually a valid number for a particular region is not performed. This can be done
    * separately with {@link #isValidNumber}.
    *
+   * <p> Note this method canonicalizes the phone number such that different representations can be
+   * easily compared, no matter what form it was originally entered in (e.g. national,
+   * international). If you want to record context about the number being parsed, such as the raw
+   * input that was entered, how the country code was derived etc. then call {@link
+   * #parseAndKeepRawInput} instead.
+   *
```

Comment:
It is here, see the next line 3027. It is just here as a parameter comment, which we don't have in C++